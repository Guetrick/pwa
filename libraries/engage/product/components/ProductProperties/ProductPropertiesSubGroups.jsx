import React from 'react';
import PropTypes from 'prop-types';
import { getSubgroupsFromProperties } from './helpers/getSubgroupsFromProperties';
import { ProductPropertiesRowGrouped } from './ProductPropertiesRowGrouped';
import { ProductPropertiesRowGroup } from './ProductPropertiesRowGroup';

/**
 * @param {Object} props The component props.
 * @return {JSX}
 */
export function ProductPropertiesSubGroups({ properties }) {
  const subGroups = getSubgroupsFromProperties(properties);

  return subGroups.map(group => (
    <React.Fragment key={group}>
      <ProductPropertiesRowGroup group={group} />
      <ProductPropertiesRowGrouped properties={properties} group={group} />
    </React.Fragment>
  ));
}

ProductPropertiesSubGroups.propTypes = {
  properties: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};
