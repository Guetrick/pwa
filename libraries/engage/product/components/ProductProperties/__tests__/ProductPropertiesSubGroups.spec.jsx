import React from 'react';
import { shallow } from 'enzyme';
import { ProductPropertiesSubGroups } from '../ProductPropertiesSubGroups';

const properties1 = [
  { label: 'test1' },
  { label: 'test2' },
];

const properties2 = [
  {
    label: 'test1',
    subDisplayGroup: 'Test',
  },
  {
    label: 'tes21',
    subDisplayGroup: 'Test',
  },
];

describe('<ProductPropertiesSubGroups />', () => {
  it('should not render if there are not subgroups defined', () => {
    const wrapper = shallow(<ProductPropertiesSubGroups properties={properties1} />);
    expect(wrapper.instance()).toEqual(null);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a group title and a group', () => {
    const wrapper = shallow(<ProductPropertiesSubGroups properties={properties2} />);
    expect(wrapper.find('ProductPropertiesRowGroup').length).toBe(1);
    expect(wrapper.find('ProductPropertiesRowGroup').prop('group')).toEqual('Test');
    expect(wrapper.find('ProductPropertiesRowGrouped').prop('group')).toEqual('Test');
    expect(wrapper.find('ProductPropertiesRowGrouped').prop('properties')).toEqual(properties2);
    expect(wrapper).toMatchSnapshot();
  });
});
