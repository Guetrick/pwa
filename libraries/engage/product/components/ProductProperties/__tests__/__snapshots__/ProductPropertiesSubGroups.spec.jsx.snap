// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPropertiesSubGroups /> should not render if there are not subgroups defined 1`] = `null`;

exports[`<ProductPropertiesSubGroups /> should render a group title and a group 1`] = `
<Fragment
  key="Test"
>
  <ProductPropertiesRowGroup
    group="Test"
  />
  <ProductPropertiesRowGrouped
    group="Test"
    properties={
      Array [
        Object {
          "label": "test1",
          "subDisplayGroup": "Test",
        },
        Object {
          "label": "tes21",
          "subDisplayGroup": "Test",
        },
      ]
    }
  />
</Fragment>
`;
